sdk/tests/deqp/modules/shared/glsShaderExecUtil.js

```diff
@@ -446,7 +446,11 @@ goog.scope(function() {
             else if (gluShaderUtil.isDataTypeMatrix(basicType)) {
                 /** @type {number} */ var numRows = gluShaderUtil.getDataTypeMatrixNumRows(basicType);
                 /** @type {number} */ var numCols = gluShaderUtil.getDataTypeMatrixNumColumns(basicType);
-                /** @type {number} */ var stride = numRows * numCols * 4;//sizeof(float);
+                // A matrix consists of several (column-major) vectors. A buffer is created for
+                // every vector in gluDrawUtil.draw() below. Data in every buffer will be tightly
+                // packed. So the stride should be 0. This is different from the code in native
+                // deqp, which use only one buffer for a matrix, the data is interleaved.
+                /** @type {number} */ var stride = 0;
```

Comment:
Excellent work @Richard-Yunchao tracking down this discrepancy between the native code and the JavaScript port. I don't know how difficult it would be, but do you think it would be worth re-porting the code from scratch from the C++ code? If so, do you have an estimate for how much work it would be? We could potentially ask Mobica to revisit it.